### PR TITLE
Update map coloring example

### PR DESCRIPTION
**Pull request by JoelPasvolsky**
I might make some minor changes to the first two paragraphs of Solution Steps after rebasing when https://github.com/dwavesystems/dwave-ocean-sdk/pull/165 is merged.

### CODE REVIEW

**Comment by JoelPasvolsky** [comment]
Just noting that the doctest failure will be fixed when dwave-system includes https://github.com/dwavesystems/dwave-system/pull/429 (release > 1.10)

**Comment by randomir** [comment]
> Just noting that the doctest failure will be fixed when dwave-system includes [dwavesystems/dwave-system#429](https://github.com/dwavesystems/dwave-system/pull/429) (release > 1.10)

Actually, doctest that's failing is this one:
```
Document: examples/map_coloring
-------------------------------
**********************************************************************
File "examples/map_coloring.rst", line 284, in default
Failed example:
    if best.energy > 0:
        print("Failed to color map. Try sampling again.")
Exception raised:
    Traceback (most recent call last):
      File "/usr/local/lib/python3.7/doctest.py", line 1337, in __run
        compileflags, 1), test.globs)
      File "<doctest default[0]>", line 1, in <module>
        if best.energy > 0:
    NameError: name 'best' is not defined
**********************************************************************
```

File: docs/examples/map_coloring.rst [hunk 1]
```diff
@@ -115,118 +134,187 @@ colors or use different maps). Canada's 13 provinces are denoted by postal codes
...
+Set a one-hot constraint on the four binary variables representing the possible
+colors for each province.
+
+>>> bqm_one_color = BinaryQuadraticModel('BINARY')
 >>> for province in provinces:
-...    variables = [province+str(i) for i in range(colors)]
-...    csp.add_constraint(one_color_configurations, variables)
-...
->>> # Add constraint that each pair of nodes with a shared edge not both select one color
+...   variables = [province + "_" + c for c in colors]
```
Comment (randomir) [inline docs/examples/map_coloring.rst:198]:
```suggestion
...   variables = [f"{province}_{c}" for c in colors]
```

File: docs/examples/map_coloring.rst [hunk 2]
```diff
@@ -115,118 +134,187 @@ colors or use different maps). Canada's 13 provinces are denoted by postal codes
...
+  3  0  1    0.0       1
+  2  1  1    1.0       1
+  ['BINARY', 4 rows, 4 samples, 2 variables]
+
+The code below sets quadratic coefficients to :math:`+1` for interactions between
+neighboring provinces' variables representing the same color, for example
+:code:`AB_y` and :code:`BC_y` (yellow for Alberta and British Columbia).
+
+>>> bqm_neighbors  = BinaryQuadraticModel('BINARY')
 >>> for neighbor in neighbors:
```
Comment (randomir) [inline docs/examples/map_coloring.rst:249]:
```suggestion
>>> for v, u in neighbors:
```

File: docs/examples/map_coloring.rst [hunk 3]
```diff
@@ -115,118 +134,187 @@ colors or use different maps). Canada's 13 provinces are denoted by postal codes
...
-...    v, u = neighbor
-...    for i in range(colors):
-...       variables = [v+str(i), u+str(i)]
-...       csp.add_constraint(not_both_1, variables)
+...   v, u = neighbor
+...   interactions = [(v + "_" + c, u + "_" + c) for c in colors]
+...   for interaction in interactions:
+...      bqm_neighbors.add_quadratic(interaction[0], interaction[1], 1)
```
Comment (randomir) [inline docs/examples/map_coloring.rst:253]:
```suggestion
...   interactions = [(f"{v}_{c}", f"{u}_{c}") for c in colors]
...   for i, j in interactions:
...      bqm_neighbors.add_quadratic(i, j, 1)
```

**Review by randomir — APPROVED** [review]
LGTM.

**Comment by JoelPasvolsky** [comment]
> Actually, doctest that's failing is this one:
> 
Now it's failing the way it should

**Pull request merged**